File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_dict = {}
        
        for i, n in enumerate(nums):
            complement = target - n
            
            if complement in nums_dict:
                return [i, nums_dict[complement]]
            nums_dict[n] = i

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    numsObj = {}
    
    for (let i in nums) {
        let complement = target - nums[i]
        
        if (complement in numsObj) {
            return [i, numsObj[complement]]
        }
        numsObj[nums[i]] = i;
    }
};